src/app.js

```diff
@@ -355,17 +378,25 @@ class Messenger extends EventEmitter {
     return this.cache.set(session._key, session);
   }
 
-  send(recipientId/*: string|number */, messageData/*: Object */, pageId/*: string|void */)/* Promise<Object> */ {
+  send(arg1/*: string|number */, arg2/*: string|number|Object */, arg3/*: Object|void */)/* Promise<Object> */ {
```

Comment:
Since this is going to be refactored into simpler logic, have you considered making the new paradigm a wrapper function for the time being and then we replace `send` down the line? This would also mean changing the third arg to the `pageAccessToken` rather than the ID, but I think that's a fair tradeoff since that is currently an optional arg and it would reduce LOC.

```
pageSend(pageId, recipientId, messageData) {
  const pageAccessToken = this.pages[pageId];
  return this.send(recipientId, messageData, pageAccessToken);
}
```

```
send(recipientId, messageData, accessToken) {
  // kills ~10 LOC here
  const pageAccessToken = accessToken || config.get('messenger.pageAccessToken');
  const options = {... }
  debug('message.send: %j', options);
  return reqPromise.post(options)
    ...
}
```
Later on, the contents of `send` can be merged into `pageSend` or `send` can be replace by `pageSend`.

Both approaches are a bit awkward TBH.  I find the above alternative easier to follow and to me, it communicates more about where the code is headed. It does come with downsides though. In either case, I think some additional comments or a link to an issue describe the next steps would provide some useful context.

*TL;DR* - I'm okay with this even though it is a bit tough to read, but wanted to propose an alternative as a thought exercise.